Standards/scs-0114-v1-secure-connections.md

```diff
@@ -0,0 +1,172 @@
+
+### Database Connections
+
+- The database servers used by the OpenStack services MUST be configured for TLS transport.
+- All OpenStack services MUST have TLS configured for the database connection via the `ssl_ca` directive. See [OpenStack service database configuration](https://docs.openstack.org/security-guide/databases/database-access-control.html#openstack-service-database-configuration).
+- Database user accounts for the OpenStack services SHOULD be configured to require TLS connections via the `REQUIRE SSL` SQL directive. See [Require user accounts to require SSL transport](https://docs.openstack.org/security-guide/databases/database-access-control.html#require-user-accounts-to-require-ssl-transport).
+- Security MAY be further enhanced by configuring the OpenStack services to use X.509 client certificates for database authentication. See [Authentication with X.509 certificates](https://docs.openstack.org/security-guide/databases/database-access-control.html#authentication-with-x-509-certificates).
+
+### Message Queue Connections
+
+- If using RabbitMQ or Qpid as the message queue service, the SSL functionality of the message broker MUST be enabled and used by the OpenStack services. See [Messaging transport security](https://docs.openstack.org/security-guide/messaging/security.html#messaging-transport-security).
+  - If using RabbitMQ, the OpenStack services' oslo.messaging configuration for RabbitMQ MUST use the `kombu_ssl_*` options accordingly to enable SSL.
+  - If using Qpid, the OpenStack services' oslo.messaging configuration for Qpid MUST set the `qpid_protocol` option to `ssl` to enable SSL.
+- If using Apache Kafka, the server listener MUST be configured to accept SSL connections. See [Apache Kafka Listener Configuration](https://kafka.apache.org/documentation/#listener_configuration).
+  - The OpenStack services' oslo.messaging configuration for Kafka MUST specify `security_protocol` as either `SSL` or `SASL_SSL` and the related options appropriately. See [Kafka Driver Options](https://docs.openstack.org/oslo.messaging/latest/admin/kafka.html#driver-options).
+
+### Hypervisor and Live Migration Connections
+
+- If using libvirt on compute nodes, the libvirt port (as per `listen_addr`) MUST NOT be exposed to the network in an unauthenticated and unprotected fashion. It SHOULD be bound to `127.0.0.1`.
+- If QEMU and libvirt are used as the hypervisor interface in Nova, QEMU-native TLS SHOULD be used. See [Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html).
+
+
+### External VM Connections
+
+- As an OPTIONAL measure to assist customers in protecting external connections to their OpenStack networks and VMs, the Neutron VPNaaS service MAY be integrated into the infrastructure and the Neutron VPNaaS API extension offered to users. See the [Neutron VPNaaS documentation](https://docs.openstack.org/neutron-vpnaas/latest/).
```

Comment:
Would it be okay to "allow" also other VPN-Endpoints? As long as this part of the standard is optional I would not see problem in allowing also other mechanisms.

Reply:
I rephrased the section in a more generic manner and mention Neutron's VPNaaS as an example now.